src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -104,20 +103,30 @@ InvokerFactory MakeMlirFwdInvokerFactory(const ConvolutionContext& ctx)
                             out_dims,
                             out_strides);
 
+    MlirConvArgs args =
+        MakeMlirConvArgs(in_dims, in_strides, weights_dims, weights_strides, out_dims, out_strides);
+
     return [=](const std::vector<Kernel>& kernels) {
-        return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+        return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) mutable {
             const auto& data_ctx = primitive_parameters.CastTo<conv::DataInvokeParams>();
             const auto& tensors  = data_ctx.tensors;
 
-            MlirConvArgs args = MakeMlirConvArgs(tensors.in,
-                                                 in_dims,
-                                                 in_strides,
-                                                 tensors.w,
-                                                 weights_dims,
-                                                 weights_strides,
-                                                 tensors.out,
-                                                 out_dims,
-                                                 out_strides);
+            // Filter
+            args.filter.basePtr =
+                const_cast<void*>(tensors.w); // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            args.filter.data =
+                const_cast<void*>(tensors.w); // NOLINT (cppcoreguidelines-pro-type-const-cast)
+
+            // Input
+            args.input.basePtr =
+                const_cast<void*>(tensors.in); // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            args.input.data =
+                const_cast<void*>(tensors.in); // NOLINT (cppcoreguidelines-pro-type-const-cast)
```

Comment:
@atamazov I'm afraid const_cast is unavoidable because:
 - The definition of `StridedMemRef` mark T not as a const pointer
   - I don't see a benefit to expand the `StridedMemRef` with a const model. 
 - MIOpen's DataInvokeParams mark input and filter as `ConstData_t`

Reply:
Just in case: MIOpen uses ConstData_t for input tensors (buffers) and Data_t for output buffer. This is not important when you build MIOpen with HIP backend, but important for OpenCL backend.

Reply:
>  This is not important when you build MIOpen with HIP backend, but important for OpenCL backend.

The const only applies to HIP. In OpenCL, `void*` is a `cl_mem` object so the extra `const` only applies to the pointer and not the data.

Reply:
Oops, correct, I've messed up things :facepalm: